src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -125,6 +128,32 @@ public SafeX509ChainHandle SafeHandle
             }
         }
 
+        private static SafeX509StoreHandle GetTrustStore(
+            X509ChainTrustMode trustMode,
+            X509Certificate2Collection customTrustStore,
+            SafeX509StackHandle untrusted,
+            SafeX509StackHandle systemTrust)
+        {
+            if (trustMode == X509ChainTrustMode.CustomRootTrust)
+            {
+                SafeX509StoreHandle x509Chain;
+                using (SafeX509StackHandle customTrust = Interop.Crypto.NewX509Stack())
+                {
+                    foreach (X509Certificate2 cert in customTrustStore)
+                    {
+                        SafeX509StackHandle toAdd = cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData) ? customTrust : untrusted;
+                        AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, toAdd);
+                    }
+
+                    x509Chain = Interop.Crypto.X509ChainNew(customTrust, SafeX509StackHandle.InvalidHandle);
+                }
+
+                return x509Chain;
```

Comment:
Why define and return this outside of the using, rather than just doing `return Interop.Crypto.X509ChainNew(...);` inside the using?